cmd/kubeapps-apis/server/packages_test.go

```diff
@@ -0,0 +1,580 @@
+
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackageSummaries: []*corev1.AvailablePackageSummary{
+					_test.MakeAvailablePackageSummary("pkg-2", mockedPackagingPlugin2.plugin),
+				},
+				Categories:    []string{"cat-1"},
+				NextPageToken: "4",
+			},
+			statusCode: codes.OK,
+		},
+		{
+			name: "it should successfully call and paginate (last page) the core GetAvailablePackageSummaries operation",
+			configuredPlugins: []*pkgsPluginWithServer{
+				mockedPackagingPlugin1,
+				mockedPackagingPlugin2,
+			},
+			request: &corev1.GetAvailablePackageSummariesRequest{
+				Context: &corev1.Context{
+					Cluster:   "",
+					Namespace: globalPackagingNamespace,
+				},
+				PaginationOptions: &corev1.PaginationOptions{PageToken: "3", PageSize: 1},
+			},
+
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackageSummaries: []*corev1.AvailablePackageSummary{
+					_test.MakeAvailablePackageSummary("pkg-2", mockedPackagingPlugin2.plugin),
+				},
+				Categories:    []string{"cat-1"},
+				NextPageToken: "4",
+			},
+			statusCode: codes.OK,
+		},
+		{
+			name: "it should successfully call and paginate (last page+1) the core GetAvailablePackageSummaries operation",
+			configuredPlugins: []*pkgsPluginWithServer{
+				mockedPackagingPlugin1,
+				mockedPackagingPlugin2,
+			},
+			request: &corev1.GetAvailablePackageSummariesRequest{
+				Context: &corev1.Context{
+					Cluster:   "",
+					Namespace: globalPackagingNamespace,
+				},
+				PaginationOptions: &corev1.PaginationOptions{PageToken: "4", PageSize: 1},
+			},
+
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackageSummaries: []*corev1.AvailablePackageSummary{},
+				Categories:                []string{"cat-1"},
+				NextPageToken:             "",
+			},
+			statusCode: codes.OK,
+		},
+		{
+			name: "it should fail when calling the core GetAvailablePackageSummaries operation with one of the plugin failing",
+			configuredPlugins: []*pkgsPluginWithServer{
+				mockedPackagingPlugin1,
+				mockedFailingPackagingPlugin,
+			},
+			request: &corev1.GetAvailablePackageSummariesRequest{
+				Context: &corev1.Context{
+					Cluster:   "",
+					Namespace: globalPackagingNamespace,
+				},
+			},
+
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackageSummaries: []*corev1.AvailablePackageSummary{},
+				Categories:                []string{""},
+			},
+			statusCode: codes.NotFound,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			server := &packagesServer{
+				plugins: tc.configuredPlugins,
+			}
+			availablePackageSummaries, err := server.GetAvailablePackageSummaries(context.Background(), tc.request)
+
+			if got, want := status.Code(err), tc.statusCode; got != want {
+				t.Fatalf("got: %+v, want: %+v, err: %+v", got, want, err)
+			}
+
+			if tc.statusCode == codes.OK {
+				opt1 := ignoreUnexportedOpts
```

Comment:
Fine as is, but really no need for `opt1`? (you can just pass in `ignoreUnexportedOpts` below).